matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/auth/DefaultAuthenticationService.kt

```diff
@@ -382,11 +382,16 @@ internal class DefaultAuthenticationService @Inject constructor(
         return getWellknownTask.execute(
                 GetWellknownTask.Params(
                         domain = matrixId.getDomain(),
-                        homeServerConnectionConfig = homeServerConnectionConfig
+                        homeServerConnectionConfig = homeServerConnectionConfig.orWellKnownDefaults()
                 )
         )
     }
 
+    private fun HomeServerConnectionConfig?.orWellKnownDefaults() = this ?: HomeServerConnectionConfig.Builder()
```

Comment:
I wasn't too sure about this, I had originally planned to make the [service entry point](https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/auth/AuthenticationService.kt#L102) non null but it feels strange to pass the `matrix id` and a homeserver config (with the url already calculated from the matrix id) 

Ideally the `getWellKnownData` could take the certificate parts of the homeserver config rather than the entire model

Reply:
I also tried with non-null parameter in my local but it produces ugly code as so said.

Reply:
I'm happy to add a build method and then remove the `matrixId` from the `getWellKnownData` :+1:

Reply:
I had a go at using the builder this however the direct usage of Android's `Uri` breaks the OnboardingViewModel tests
https://github.com/vector-im/element-android/compare/feature/adm/matrix-id-via-builder

will merge _as is_ for the time being